### ISSUE TITLE
No doc inheritance for enum values.

### ISSUE THREAD

**Issue by adaskos-signal** [saucecontrol/InheritDoc#14]
Greetings.
First of all thanks for your work in this library. It's exactly what I've been looking for, and it's working very well in my preliminary tests.

I've only noticed an issue with `enum` values. They don't seem to work, unless their `cref` target is referenced by chance by another property, in which case it works well.

I've dived into your code to understand what I was doing wrong. 
It seems to me that because you rely on `Methods` to collect `cref` references, you miss `enum` values which are `Fields`, and they don't end up in `refDocs` when you check if `refCref.Contains(name)`.

Is my understanding correct? If so, is there a plan to add support for `enum` values?
Or maybe a workaround you can suggest? I can only think of a dummy class with dummy properties, but it's a bit messy and manual.

Could you avoid the above issue by simply removing the check for `refCref.Contains(name)`? Will this break something? It would let all `members` be available for replacement, in the next steps.

**Comment by adaskos-signal** [comment]
Yes, my case is that of a `cref` towards a different assembly (a different project in my solution. or a NuGet library).

I see your point about being frugal. 
But as far as I can read your code, you replace `inheritdocs` elements already found in the comments. So you would only be fetching documentation from explicit references. It's only that now you have discarded their target.

In which scenario would there be unwanted replacements?

**Comment by saucecontrol** [comment]
Part of the motivation behind the way it's written is that the collected docs of all references would be huge, and the processor combines them all into a single doc to use for lookups.  By ignoring any docs that are not going to be used, we keep the size of that combined doc down, which limits memory use and makes the searches for docs within that set faster.

`refCref` also specifically excludes any docIDs defined by your own project so that we know not to look for those in the referenced library docs.  Since the search for replacements is done based on docID, any conflicting names could result in docs coming from an external reference when they should have come from the source project.

As far as the logic of only iterating methods (not fields) in the project's types, only methods can have an inheritance defined by hierarchy.  Fields can't be overridden (only hidden), so anything else should require an explicit `cref`.

I'm not sure why your particular case is failing with the logic in place, but if you can put together a simple repro (preferably using inheritance from something in the BCL), I can give it a look.

**Comment by adaskos-signal** [comment]
I'll try to come up with a unit test.

**Comment by adaskos-signal** [comment]
I pushed a [commit](https://github.com/adaskos-signal/InheritDoc/commit/760c17233cfefbcc0a880fe5e9f3830ff157f443?diff=split&w=0) to my fork, showcasing the failing scenario.

Of the five new tests, the two with internal reference (same project) pass already. 
If I reference other assemblies (`System` or a new library project I added for the test), they don't pull the summary.

**Comment by saucecontrol** [comment]
Thanks for that!  I see the problem now.  You're correct, it's not evaluating fields at all when building the doc map, except for the purpose of trimming private/internal members.

The following also fails to resolve:

```c#
/// <summary>Class C</summary>
public class C
{
	/// <inheritdoc cref="String.Empty" />
	public string F;
}
```

So the fix is relatively simple.  I'll just need to add fields to the `docMap` as is done for methods so their external `cref`s are also included in `refCref`.

**Comment by saucecontrol** [comment]
Fixed in https://github.com/saucecontrol/InheritDoc/commit/d4ddea5a92884931c3b71e3fa211039ca19515e0

**Comment by adaskos-signal** [comment]
Nice! Thanks for the quick turnaround.

**Comment by adaskos-signal** [comment]
Hi again. 
Not sure what your release practice is. 
Should I expect a new NuGet version for the above fix? Or build locally for now?

**Comment by saucecontrol** [comment]
Yeah, I plan on snapping a release with this fix next week.  I like to let things simmer for a few days in case I think of something else I'd like to get in to the new version.  I also find that issue reports tend to come in bunches for some reason, and I always hate it when I publish a release and someone reports a new issue the next day 😅

There's a [nuget feed for the CI builds](https://dev.azure.com/saucecontrol/InheritDoc/_artifacts/feed/inheritdoc_ci) if you want to pick it up from there in the meantime.
